src/esm_runscripts/namelists.py

```diff
@@ -491,7 +492,18 @@ def apply_iceberg_calving(config):
                     ib_num_old = 0
 
                 print(" * iceberg_dir = ", config["fesom"].get("iceberg_dir"))
-                ib_num_new = sum(1 for line in open(config["fesom"].get("iceberg_dir") + "/LON.dat"))
+                iceberg_dir = config["fesom"].get("iceberg_dir", None)
+                if os.path.isfile(iceberg_dir + "/LON.dat"):
+                    ib_num_new = sum(1 for line in open(config["fesom"].get("iceberg_dir") + "/LON.dat"))
+                elif config["general"]["check"]:
+                    # MA: Get the number of icebergs from the path for check runs where the file is not
+                    # available (e.g. CI in GitHub)
+                    ib_num_new = int([x for x in iceberg_dir.split("/") if len(x)>0][-1].split("_")[-1])
```

Comment:
Sorry but...em....what? I trust you that this works, but what wizardry is this? You are counting the number of non-empty path elements, but then the last one, and then the last one after splitting the `_`? 

Even if it never breaks, someone at some point will stumble across this code and be highly confused. Just put a comment or something that shows what number this gets. I assume it is something like:
```
/some///./crazy/path/with///a/new/file_38125
```
which yields
```
38125
```

Reply:
Rather than using string functions we should use Pathlib or os.path methods. Eg. `parent`, ...

Reply:
The minimum would have been to use regex, but I don't know what I was doing that day. In any case, I don't like this either. I will probably revert most of the `namelists.py` changes and fix it from the resources side (avoiding being so so invasive to the ESM-Tools code)

Reply:
Assuming the number is always the last thing in the file name:
```suggestion
                    ib_num_new = int(iceberg_dir.split("_")[-1])
```

Reply:
solved by reverting the changes in `namelist.py` and handling all throught esm_tests